### PR TITLE
ADM-914:[backend][frontend] feat: set file name with time range when generate file

### PR DESCRIPTION
**Pull request by yulongcai**
## Summary
set file name with time range when generate file

## Before
before
## After
Snipaste_2024-04-17_16-13-14

## Note

_Null_

### CODE REVIEW

File: backend/src/test/java/heartbeat/service/report/KanbanCsvServiceTest.java [hunk 1]
```diff
@@ -89,6 +89,9 @@ void shouldSaveCsvWithoutNonDoneCardsWhenNonDoneCardIsNull() throws URISyntaxExc
 						.targetFields(List.of(TargetField.builder().name("Doing").build(),
 								TargetField.builder().name("Done").build()))
 						.build())
+					.csvTimeStamp("1720000000000")
+					.startTime("1710000000000")
+					.endTime("1712678399999")
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/KanbanCsvServiceTest.java:94]: convert `1720000000000` all those value into a readable variable

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java [hunk 2]
```diff
@@ -71,31 +71,41 @@ public class ReportServiceTest {
...
+				() -> reportService.exportCsv(ReportType.METRIC, String.valueOf(invalidTimestamp), "", ""));
+		verify(csvFileGenerator, never()).getDataFromCSV(ReportType.METRIC, mockTimeRangeTimeStamp);
 	}
 
 	@Nested
 	class GenerateReportByType {
 
 		String timeStamp = "1683734399999";
 
+		String startTimeStamp = "1710000000000";
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:101]: 用年月日转换，后面的测试都改

File: backend/src/main/java/heartbeat/controller/report/ReportController.java [hunk 3]
```diff
@@ -37,15 +38,15 @@ public class ReportController {
...
+	@GetMapping("/{reportType}/{timeStamp}")
 	public InputStreamResource exportCSV(
 			@Schema(type = "string", allowableValues = { "metric", "pipeline", "board" },
 					accessMode = Schema.AccessMode.READ_ONLY) @PathVariable() ReportType reportType,
-			@PathVariable String filename) {
-		log.info("Start to export CSV file_reportType: {}, filename: {}", reportType.getValue(), filename);
-		InputStreamResource result = reportService.exportCsv(reportType, Long.parseLong(filename));
-		log.info("Successfully get CSV file_reportType: {}, filename: {}, _result: {}", reportType.getValue(), filename,
-				result);
+			@PathVariable String timeStamp, @Parameter String startDate, @Parameter String endDate) {
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/controller/report/ReportController.java:45]: put those 2 new variables into the API path

image

File: frontend/src/clients/report/CSVClient.ts [hunk 4]
```diff
@@ -9,7 +9,13 @@ export class CSVClient extends HttpClient {
 
   exportCSVData = async (params: CSVReportRequestDTO) => {
     await this.axiosInstance
-      .get(`/reports/${params.dataType}/${params.csvTimeStamp}`, { responseType: 'blob' })
+      .get(`/reports/${params.dataType}/${params.csvTimeStamp}`, {
+        params: {
+          startDate: this.parseCollectionDateToHumanDate(params.startDate),
+          endDate: this.parseCollectionDateToHumanDate(params.endDate),
+        },
```
Comment (guzhongren) [inline frontend/src/clients/report/CSVClient.ts:16]: fix with the API scheme

File: frontend/src/clients/report/CSVClient.ts [hunk 5]
```diff
@@ -9,7 +9,13 @@ export class CSVClient extends HttpClient {
 
   exportCSVData = async (params: CSVReportRequestDTO) => {
     await this.axiosInstance
-      .get(`/reports/${params.dataType}/${params.csvTimeStamp}`, { responseType: 'blob' })
+      .get(`/reports/${params.dataType}/${params.csvTimeStamp}`, {
```
Comment (guzhongren) [inline frontend/src/clients/report/CSVClient.ts:12]: No frontend test??

**Review by guzhongren — COMMENTED** [review]
Change with the comments

File: backend/src/test/java/heartbeat/service/report/KanbanCsvServiceTest.java (hunk 1 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/KanbanCsvServiceTest.java]: done

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 2 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java]: done

File: backend/src/main/java/heartbeat/controller/report/ReportController.java (hunk 3 above)
Comment (yulongcai) [reply backend/src/main/java/heartbeat/controller/report/ReportController.java]: done

File: frontend/src/clients/report/CSVClient.ts (hunk 5 above)
Comment (TingyuDong) [reply frontend/src/clients/report/CSVClient.ts]: Resolved.

**Comment by guzhongren** [comment]
下午两个团队一起diff一哈

**Review by Leiqiuhong — APPROVED** [review]

**Pull request merged**